src/dataflow/src/source/mod.rs

```diff
@@ -1857,6 +1878,59 @@ impl<K: Codec, V: Codec> PersistentTimestampBindingsConfig<K, V> {
     }
 }
 
+fn maybe_emit_timestamp_bindings(
+    source_name: &str,
+    worker_id: usize,
+    timestamp_bindings_updater: &mut Option<TimestampBindingUpdater>,
+    timestamp_histories: &mut TimestampBindingRc,
+    consistency_info: &ConsistencyInfo,
+    bindings_cap: &Capability<u64>,
+    bindings_output: &mut OutputHandle<
+        u64,
+        ((SourceTimestamp, AssignedTimestamp), u64, isize),
+        Tee<u64, ((SourceTimestamp, AssignedTimestamp), u64, isize)>,
+    >,
+) {
+    let timestamp_bindings_updater = match timestamp_bindings_updater.as_mut() {
+        Some(updater) => updater,
+        None => {
+            return;
+        }
+    };
+
+    let changes = timestamp_bindings_updater.update(timestamp_histories);
+
+    // Emit required changes downstream.
+    let to_emit = changes
+        .into_iter()
+        .filter(|((source_ts, _assigned_ts), _diff)| {
+            consistency_info.responsible_for(&source_ts.partition)
```

Comment:
This is left over from when we didn't. I'll check again and see if we still need it. I might upgrade it to panic, in case we have something that we're not responsible for, as a sanity check.

Reply:
did you still want to upgrade this to a panic?